### PR TITLE
Remove no longer needed Emscripten metadata.

### PR DESCRIPTION
**Pull request by PiotrSikora**
Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

File: src/wasm.cc [hunk 1]
```diff
@@ -302,36 +286,6 @@ bool WasmBase::initialize(const std::string &code, bool allow_precompiled) {
     if (!ok) {
       return false;
     }
-    auto metadata = wasm_vm_->getCustomSection("emscripten_metadata");
```
Comment (mathetake) [inline src/wasm.cc:305]: nit: looks like we no longer need to expose getCustomSection in WasmVm class, although we may need them later?

**Review by mathetake — APPROVED** [review]
yeah, this was what I was thinking about. Just curious, what's the original purpose of getting emscripten information?

File: src/wasm.cc (hunk 1 above)
Comment (PiotrSikora) [reply src/wasm.cc:305]: It's used to extract precompiled modules in V8 (which we could make internal), and in some tests in Envoy.

**Comment by PiotrSikora** [comment]
We needed it to get memory offsets to initialize some pointers at startup, but nowadays that's handled by crt1 constructors emitted by Emscripten.

**Pull request merged**